src/net.cpp

```diff
@@ -2865,3 +2867,23 @@ uint64_t CConnman::CalculateKeyedNetGroup(const CAddress& ad) const
 
     return GetDeterministicRandomizer(RANDOMIZER_ID_NETGROUP).Write(vchNetGroup.data(), vchNetGroup.size()).Finalize();
 }
+
+void LogMessage(const CNode& node, const std::string& command, const std::vector<unsigned char>& data, bool is_incoming)
+{
+    int64_t time = GetTimeMicros();
```

Comment:
Q: Is it important that this time is equal to the receive time of the message?

If yes, you could pass in the time from the caller. (msg.m_time)

Reply:
The exact time doesn't matter, this is mainly used for sorting the messages later.  All that's important is that messages that are sent later have a later time.  To avoid races I only call LogMessage from the MessageHandler thread.  When sending a message in PushMessage, we don't have anything like msg.m_time until the message is into SocketHandler.